### DISCUSSION TITLE
Bare Metal upgrade instructions

### DISCUSSION THREAD

**Discussion by skorvek** [jonaswinkler/paperless-ng#1095] [Support]
Hello and again thank a ton for this application. 

I am running a bare-metal install on a Ubuntu 20.04 LTS VM image and am trying to upgrade from 1.4.3->1.4.4.  The upgrade instructions are minimal so not sure if I'm doing it entirely correctly.  The webserver isn't starting after the upgrade because I don't have a .local directory to contain the required binaries. The consumer and scheduler started fine.

So the instructions say to grab and unpack the latest release.  I obviously don't want to overwrite my config, etc- so I downloaded the tarball to my use directory, moved my existing /opt/paperless directory to /opt/paperless-1.4.3 and expanded the tarball to /opt/paperless. I changed ownership to the paperless user, copied back the config file, migrated the database schema, and did the pip requirements install- which seemed to try to reinstall everything again but did not pull down the gunicorn binaries.

I'm not using ansible yet (future project) - am I doing this correctly? I assume it's something with the python environment. This is a standalone server with a nginx proxy sitting on another machine. Sole purpose is to run paperless, so I don't have any custom python environments setup. 

A few more details- not a python developer, but able to read documentation. I reviewed the initial setup instructions- had forgotten to run pip as the paperless user - so I did that. Created the .local directory but instead created empty python, python3.8 etc directories that I didn't have before. Running requirements didn't do anything- noting that pip had already installed the dependencies in the /user/local tree... not where they should be. Running with that, I adjusted the sysctl daemon script to run gunicorn from there instead of local - but won't because the permission in the default dist-config directories for these dependencies is root:staff.   Clearly I'm missing something here with pip but can't find anything based upon the documentation here which worked fine on initial install.

So in the meantime, I rolled back pointing /opt/paperless to /opt/paperless-1.4.3. but the DB had been upgraded....   will that break anything?

**Comment by steviehs** [comment]
For upgrading bare metal I was successful with:

stop paperless services

cp -r /opt/paperless /opt/paperless.bak
unpack new paperless-ng archive to /opt/paperless

do all python magic:
sudo -Hu paperless pip3 install --upgrade pip
sudo -Hu paperless pip3 install --upgrade -r requirements.txt

copy or adapt paperless.conf from .bak dir

migrate database:
sudo -Hu paperless python3 manage.py migrate

start paperless services
